djvm/src/main/kotlin/net/corda/djvm/rules/implementation/DisallowNonDeterministicMethods.kt

```diff
@@ -77,6 +93,13 @@ object DisallowNonDeterministicMethods : Emitter {
         private val hasAnyClassReflection = hasClassReflection || isNewInstance
 
         fun runFor(className: String): Choice = when {
+            isClassLoader && instruction.memberName == "<init>" -> if (instruction.descriptor == "()V") {
+                Choice.INIT_CLASSLOADER
+            } else {
+                Choice.FORBID
+            }
+            isClassLoader && instruction.memberName == "getParent" -> Choice.RETURN_NULL
```

Comment:
Assuming that i am understanding this right, i would instead consider throwing a sandbox violation exception rather than returning null. After all, the code is doing something that is forbidden.

Reply:
The DJVM's primary goal is only to execute code deterministically. I can't think of a good reason for asking for a classloader's parent, but third party libraries such as `kotlin-stdlib` weren't written with the DJVM in mind and are out of our control. I would therefore prefer to _tolerate_ cases like this where possible.

Reply:
They should realize that their code is violating the sandbox requirements, isnt returning null effectively equivalent to masking a runtime error?

Reply:
Except it might not be _their_ code! And `ClassLoader.getParent()` is allowed to return `null` because that's what classes in the bootstrap classloader do.

Reply:
Why does it matter if it is code they have written or not? 

My thought is the following:
- if my code is deterministic and secure as enforced by the sandbox constraints then i should get the same output inside and outside the sandbox
- otherwise it should just not run inside the sandbox, which means an exception should be the only possible outcome.

Reply:
What is the user supposed to do with a `RuleViolationError` from (say) `kotlin-stdlib`?